torture/src/spawn.rs

```diff
@@ -102,6 +102,10 @@ fn spawn_child_with_sock(socket_fd: RawFd) -> Result<std::process::Child> {
     }
 
     let mut cmd = Command::new(program);
+    // Override the PGID of the spawned process. The motivation for this is ^C handling. To handle
+    // ^C the shell will send the SIGINT to all processes in the process group. We are handling
+    // SIGINT manually in the supervisor process.
+    cmd.process_group(0);
```

Comment:
if you had allowed for multiple agents, each agent (child) would need to have a different `process_group`?

Reply:
Yes!

Setting the pgid ([man](https://www.man7.org/linux/man-pages/man2/setpgid.2.html)), at least on Linux, to 0 essentially creates a new process group with the ID equal to the PID of the (child) process. Hence, each child spawned here would end up in their own process group.